bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java

```diff
@@ -108,10 +108,14 @@ public int size() {
 
     /**
      * Remove an item from the map.
+     *
      * @param key key of the item to be removed
+     * @return boolean to indicate whether given key is removed.
      */
-    public void remove(K key) {
+    public boolean remove(K key) {
+        boolean hasKey = map.containsKey(key);
         map.remove(key);
+        return hasKey;
```

Comment:
if `hasKey` is false, we don't need to remove :)

Reply:
+1 fixed :)